dlang/psi-impl/src/main/kotlin/io/github/intellij/dlanguage/resolve/ScopeProcessorImplUtil.kt

```diff
@@ -22,6 +22,10 @@ object ScopeProcessorImplUtil {
             is AliasInitializer ->
                 return processor.execute(def, state)
             is AliasDeclaration -> {
+                // the element is in the alias declaration, so it can’t match to this declaration
+                if (lastParent == def)
+                    return true
+
```

Comment:
Another fix/improvement, that ensure that the value of the alias won’t resolve to the alias declaration in case of same name (example: `alias Foo = Bar.Foo`).